### PR TITLE
add bazel rule swift

### PR DESCRIPTION
**Pull request by jackyko1991**
fix issue https://github.com/tensorflow/tensorflow/issues/25747

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -47,6 +47,11 @@ http_archive(
     strip_prefix = "rules_apple-0.13.0",
     urls = ["https://github.com/bazelbuild/rules_apple/archive/0.13.0.tar.gz"],
 )
+http_archive(
```
Comment (temrich) [inline WORKSPACE]: This should not be needed. Swift rules are pulled in via Apple rules:

https://github.com/bazelbuild/rules_apple/blob/master/apple/repositories.bzl#L115

Reference: #25735.

**Comment by preethivenkatesh** [comment]
I applied this fix to my local branch and it passed successfully

**Comment by temrich** [comment]
Which version of Bazel are you using?

**Comment by preethivenkatesh** [comment]
0.20.0

**Comment by temrich** [comment]
0.22 (latest version) works for sure, and I believe 0.21 does as well. If you can, it's probably best to update to the latest version of Bazel.

**Comment by jackyko1991** [comment]
I will update to 0.22 to see if it works.

If that's the case I think documentations should be updated

**Comment by jackyko1991** [comment]
https://www.tensorflow.org/install/source

The offical documentation only require bazel 0.15,that maybe a source of problem

**Comment by preethivenkatesh** [comment]
I should mention that I don't experience this issue while using bazel 0.19.*

**Comment by temrich** [comment]
That's a good point! Will discuss with the team to see if we want to update the documentation or explicitly pull in the Swift Bazel rules via the WORKSPACE file to support older versions of Bazel. 

If we explicitly pull in the Swift Bazel rules, does that resolve the issue for 0.20 and older?

**Comment by jackyko1991** [comment]
Same issue occurs for both bazel 0.21 and 0.22. I think explicitly defining the Swift rules would be more compatible to most bazel versions

**Comment by preethivenkatesh** [comment]
actually it's not a problem with older versions, but with later versions starting from 0.20. The rules will have to be explicitly mentioned starting from versions 0.20

**Comment by temrich** [comment]
@jackyko1991 and @preethivenkatesh, which platforms are you building on? When you upgraded to Bazel 0.22, did you run `bazel clean --expunge`? I don't see the issue on MacOS or Linux using Bazel 0.22.

**Comment by preethivenkatesh** [comment]
Actually, I was testing 0.20 on **Windows** and ran into issues. However, 0.20 seems to work fine on Linux without having to mention the rules explicitly on WORKSPACE. BTW, my comment on later versions(0.21 and .22) was based on @jackyko1991 comment

**Comment by temrich** [comment]
Thank you for continuing to look into this. Do you mind providing the verbose logging for the error you are seeing on Windows (without explicitly loading the Swift rules in the WORKSPACE)?

**Comment by temrich** [comment]
#25747 was closed as it appears to be a git-related issue.

**Comment by jackyko1991** [comment]
I used the latest git and tried bazel 0.21 and 0.22 on windows 10, same error occurs if swift rule is not explicitly defined

**Comment by temrich** [comment]
@jackyko1991 can you please provide the detailed log with the error you are getting on Windows?

cc @allevato

**Comment by jackyko1991** [comment]
``` bash
WARNING: Option 'experimental_shortened_obj_file_path' is deprecated
INFO: Invocation ID: 03a58090-2db6-4b61-85db-3cab65a42a6d
ERROR: error loading package '': Encountered error while reading extension file 'swift/repositories.bzl': no such package '@build_bazel_rules_swift//swift': Traceback (most recent call last):
        File "C:/users/jacky/_bazel_jacky/2exbtrly/external/bazel_tools/tools/build_defs/repo/git.bzl", line 163
                _clone_or_update(ctx)
        File "C:/users/jacky/_bazel_jacky/2exbtrly/external/bazel_tools/tools/build_defs/repo/git.bzl", line 73, in _clone_or_update
                fail(("error cloning %s:\n%s" % (ctx....)))
error cloning build_bazel_rules_swift:
+ cd C:/users/jacky/_bazel_jacky/2exbtrly/external
+ rm -rf C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift
rm: cannot remove 'C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift': Device or resource busy
rm: cannot remove 'C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift': Device or resource busy
ERROR: error loading package '': Encountered error while reading extension file 'swift/repositories.bzl': no such package '@build_bazel_rules_swift//swift': Traceback (most recent call last):
        File "C:/users/jacky/_bazel_jacky/2exbtrly/external/bazel_tools/tools/build_defs/repo/git.bzl", line 163
                _clone_or_update(ctx)
        File "C:/users/jacky/_bazel_jacky/2exbtrly/external/bazel_tools/tools/build_defs/repo/git.bzl", line 73, in _clone_or_update
                fail(("error cloning %s:\n%s" % (ctx....)))
error cloning build_bazel_rules_swift:
+ cd C:/users/jacky/_bazel_jacky/2exbtrly/external
+ rm -rf C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift
rm: cannot remove 'C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift': Device or resource busy
rm: cannot remove 'C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift': Device or resource busy
INFO: Elapsed time: 0.826s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
    Fetching @build_bazel_rules_swift; Cloning tags/0.6.0 of https://github.com/bazelbuild/rules_swift.git
```

versions:
os: windows 10

git: 2.20.1.windows.1

bazel: 
INFO: Invocation ID: e003576b-b676-44ea-87e2-754b784655d5
Build label: 0.22.0
Build target: bazel-out/x64_windows-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Mon Jan 28 12:59:27 2019 (1548680367)
Build timestamp: 1548680367
Build timestamp as int: 1548680367

**Comment by allevato** [comment]
These errors don't look specific to anything _inside_ rules_swift; they seem to indicate a problem cloning the repository because of existing file locks or something else:

```
error cloning build_bazel_rules_swift:
+ cd C:/users/jacky/_bazel_jacky/2exbtrly/external
+ rm -rf C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift
rm: cannot remove 'C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift': Device or resource busy
rm: cannot remove 'C:/users/jacky/_bazel_jacky/2exbtrly/external/build_bazel_rules_swift': Device or resource busy
```

**Comment by jackyko1991** [comment]
I have reboot the computer and run with admin right, same error occurs

在 2019年2月28日週四 23:21，Tony Allevato <notifications@github.com> 寫道：

**Comment by preethivenkatesh** [comment]
My log is a little different. I can get around this by explicitly mentioning the swift rules

 `fatal: unable to access 'https://github.com/bazelbuild/rules_swift.git/': error setting certificate verify `

git 0.19.1 windows.1
bazel 0.20.0

Detailed log
```
WARNING: The following configs were expanded more than once: [monolithic]. For repeatable flags, repeats are counted twice and may lead to unexpected behavior.
INFO: Invocation ID: 96e7d4c7-98e5-49ba-85b5-1a90bddb1848
ERROR: error loading package '': Encountered error while reading extension file 'swift/repositories.bzl': no such package '@build_bazel_rules_swift//swift': Traceback (most recent call last):
        File "C:/users/pvenkat2/desktop/path_to_output/external/bazel_tools/tools/build_defs/repo/git.bzl", line 166
                _clone_or_update(ctx)
        File "C:/users/pvenkat2/desktop/path_to_output/external/bazel_tools/tools/build_defs/repo/git.bzl", line 72, in _clone_or_update
                fail(("error cloning %s:\n%s" % (ctx....)))
error cloning build_bazel_rules_swift:
+ cd C:/users/pvenkat2/desktop/path_to_output/external
+ rm -rf C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift
+ git clone --depth=1 https://github.com/bazelbuild/rules_swift.git C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift
Cloning into 'C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift'...
fatal: unable to access 'https://github.com/bazelbuild/rules_swift.git/': error setting certificate verify locations:
  CAfile: C:/Program Files/Git/mingw64/libexec/ssl/certs/ca-bundle.crt
  CApath: none
+ git clone https://github.com/bazelbuild/rules_swift.git C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift
Cloning into 'C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift'...
fatal: unable to access 'https://github.com/bazelbuild/rules_swift.git/': error setting certificate verify locations:
  CAfile: C:/Program Files/Git/mingw64/libexec/ssl/certs/ca-bundle.crt
  CApath: none
ERROR: error loading package '': Encountered error while reading extension file 'swift/repositories.bzl': no such package '@build_bazel_rules_swift//swift': Traceback (most recent call last):
        File "C:/users/pvenkat2/desktop/path_to_output/external/bazel_tools/tools/build_defs/repo/git.bzl", line 166
                _clone_or_update(ctx)
        File "C:/users/pvenkat2/desktop/path_to_output/external/bazel_tools/tools/build_defs/repo/git.bzl", line 72, in _clone_or_update
                fail(("error cloning %s:\n%s" % (ctx....)))
error cloning build_bazel_rules_swift:
+ cd C:/users/pvenkat2/desktop/path_to_output/external
+ rm -rf C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift
+ git clone --depth=1 https://github.com/bazelbuild/rules_swift.git C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift
Cloning into 'C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift'...
fatal: unable to access 'https://github.com/bazelbuild/rules_swift.git/': error setting certificate verify locations:
  CAfile: C:/Program Files/Git/mingw64/libexec/ssl/certs/ca-bundle.crt
  CApath: none
+ git clone https://github.com/bazelbuild/rules_swift.git C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift
Cloning into 'C:/users/pvenkat2/desktop/path_to_output/external/build_bazel_rules_swift'...
fatal: unable to access 'https://github.com/bazelbuild/rules_swift.git/': error setting certificate verify locations:
  CAfile: C:/Program Files/Git/mingw64/libexec/ssl/certs/ca-bundle.crt
  CApath: none
INFO: Elapsed time: 15.353s
INFO: 0 processes.
````

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**